### PR TITLE
Ocean Node Docs Update

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Fixes # .

Changes proposed in this PR:

- Updating the documentation to give an introduction to Ocean Node. 

Preview link here: https://oceanprotocol.gitbook.io/node-ocean-protocol-docs/4hGZcmrWEN1gIa5V0qCl/

### CODE REVIEW

**Comment by mihaisc** [comment]
this also needs update https://oceanprotocol.gitbook.io/node-ocean-protocol-docs/4hGZcmrWEN1gIa5V0qCl/developers/architecture

**Review by mariacarmina — COMMENTED** [review]
Some suggestions on my side.

File: developers/ocean-node/node-architecture.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+We can move the entire encryption/decryption/MPC process to TEE, giving us full security regarding plain data.
+
+A node that supports TEE should expose its proof in the status endpoint.
+
+With MPC encryption (both assets & files), other nodes should refuse to reassemble the private key if the node does not expose & pass the proof check.
+
+#### Database Storage Wrapper
+
+To be determined (TBD).
```
[the remark this answers is not in the document]
Comment (jamiehewitt15) [reply developers/ocean-node/node-architecture.md]: I've removed it.

File: developers/ocean-node/http-endpoints.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  * **Description**: Broadcasts a command to the network.
+* **Direct Command**
+  * **Method**: POST
+  * **Endpoint**: `/directCommand`
+  * **Description**: Sends a direct command to a specific peer.
+
+### Logging
+
+* **Logs**
+  * **Method**: POST
```
[the remark this answers is not in the document]
Comment (jamiehewitt15) [reply developers/ocean-node/http-endpoints.md:49]: Do you mean removing this page entirely and just linking to API.md instead?

[the remark this answers is not in the document]
Comment (mariacarmina) [reply developers/ocean-node/http-endpoints.md:49]: yes, that would make more sense

[the remark this answers is not in the document]
Comment (mariacarmina) [reply developers/ocean-node/http-endpoints.md]: The same for c2d phase, you got the content of the c2d readme, I think it's better to link the readme and have an intro instead for it. You can explain that for this quarter when ocean node is released (with alpha version) we will be barckwards compatible with the c2d stack that we have currently in production and devs can interact easily with c2d using ocean-cli commands. For following quarters we'll focus on c2d.2 which documentation you'll link from ocean-node.

Does it sound ok from your perspective?

File: developers/ocean-node/compute-to-data-c2d.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
+# Compute-to-data (C2D)
```
Comment (mariacarmina) [inline developers/ocean-node/compute-to-data-c2d.md:1]: Please check this comment: https://github.com/oceanprotocol/docs/pull/1447#discussion_r1691299910

File: developers/ocean-node/node-architecture.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+### Components & Modules
+
+#### Indexer
+
+An off-chain, multi-chain metadata & chain events cache. It continually monitors the chains for well-known events and caches them (V4 equivalence: Aquarius).
+
+Features:
+
+* Monitors MetadataCreated & MetadataUpdated and stores DDOs.
```
Comment (mariacarmina) [inline developers/ocean-node/node-architecture.md:57]: Monitors metadata events, we also monitor MetadataState event

File: developers/ocean-node/node-architecture.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+* Validates DDOs according to multiple SHACL schemas.
```
Comment (mariacarmina) [inline developers/ocean-node/node-architecture.md:58]: according to custom SHACL schemas provided by the user which is hosting the node.

File: developers/ocean-node/node-architecture.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+* Provides proof for valid DDOs.
+* Monitors data token contracts & stores orders.
```
Comment (mariacarmina) [inline developers/ocean-node/node-architecture.md:60]: monitors datat token contracts transactions/operations (regarding orders, minting, creating pricing schemas such as dispenser and fre)

File: developers/ocean-node/node-architecture.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+#### Provider
+
+* Performs checks on-chain for buyer permissions and payments.
+* Encrypts the URL and metadata during publishing.
+* Decrypts the URL when the dataset is downloaded or a compute job is started.
+* Encrypts/decrypts files before storage/while accessing.
+* Provides access to data assets by streaming data (and never the URL).
+* Provides compute services (connects to multiple C2D engines: light, Ocean C2D, third parties).
+
+### Modules
```
Comment (mariacarmina) [inline developers/ocean-node/node-architecture.md:72]: Should we mention performance testing and system testing somewhere else? I think it's a good idea to throw few works about them.

File: developers/ocean-node/node-architecture.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+* Monitors MetadataCreated & MetadataUpdated and stores DDOs.
+* Validates DDOs according to multiple SHACL schemas.
+* Provides proof for valid DDOs.
+* Monitors data token contracts & stores orders.
+* Allows queries for all the above.
+
+#### Provider
+
+* Performs checks on-chain for buyer permissions and payments.
+* Encrypts the URL and metadata during publishing.
```
Comment (mariacarmina) [inline developers/ocean-node/node-architecture.md:66]: We need to say some words for fees like once provider fees are paid, from initialize point, we can consume assets or starting compute jobs.

**Review by mariacarmina — COMMENTED** [review]
more detail-oriented comments

File: developers/ocean-node/compute-to-data-c2d.md (hunk 3 above)
Comment (mariacarmina) [reply developers/ocean-node/compute-to-data-c2d.md:1]: I still see here that c2d readme is not linked in this page.

**Review by mariacarmina — COMMENTED** [review]
a pending suggestion on c2d side

**Review by mariacarmina — APPROVED** [review]
Looks good to me! 
@mihaisc can you also take a final look?

File: developers/ocean-node/node-architecture.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+<figure><img src="../../.gitbook/assets/image (1).png" alt=""><figcaption><p>Ocean Node Infrastructure diagram</p></figcaption></figure>
+
+### Features
+
+* libp2p supports ECDSA key pairs, and node identity should be defined as a public key.
+* Multiple ways of storing URLs:
+  * Choose one node and use that private key to encrypt URLs (enterprise approach).
+  * Choose several nodes, so your files can be accessed even if one node goes down (given at least one node is still alive).
+  * Use MPC to divide the key among multiple nodes (so no one node has the entire key).
```
Comment (mihaisc) [inline developers/ocean-node/node-architecture.md:25]: we still have mentions of MPC and TEE, plus in the diagram

**Comment by mihaisc** [comment]
Also we are branding it as "Ocean nodes" so change it to this wherever it's possible, menu, titles, etc. Put `Ocean node` strictly when you need it

**Comment by mihaisc** [comment]
image

and under it
`Use MPC to divide the key among multiple nodes (so no one node has the entire key).`

**Review by mihaisc — APPROVED** [review]

**Pull request merged**